cmd/beaker/dataset.go

```diff
@@ -434,6 +399,82 @@ func newDatasetStreamFileCommand() *cobra.Command {
 	return cmd
 }
 
+func newDatasetSyncCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "sync <source> <target>",
+		Short: "Sync the source to an uncommitted dataset",
+		Args:  cobra.ExactArgs(2),
+	}
+
+	var concurrency int
+	cmd.Flags().IntVar(
+		&concurrency,
+		"concurrency",
+		defaultConcurrency,
+		"Number of files to upload at a time")
+
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		source := args[0]
+		target := args[1]
+
+		if err := syncDataset(source, beaker.Dataset(target), concurrency); err != nil {
+			return err
+		}
+
+		if quiet {
+			fmt.Println(target)
+		} else {
+			fmt.Println("Done.")
+		}
+		return nil
+	}
+	return cmd
+}
+
+func syncDataset(source string, target *client.DatasetHandle, concurrency int) error {
+	info, err := os.Stat(source)
+	if err != nil {
+		return err
+	}
+	if info.Mode()&(os.ModeSymlink|os.ModeNamedPipe|os.ModeSocket|os.ModeDevice) != 0 {
+		return errors.Errorf("%s is a %s", source, modeToString(info.Mode()))
```

Comment:
```suggestion
		return errors.Errorf("source %q is a %s", source, modeToString(info.Mode()))
```

But if what you have is consistent with other such messages then go for the consistency.